Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/vcs.xml	(revision )
+++ .idea/vcs.xml	(revision )
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: app/src/main/java/com/lihang/frescodemo/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/lihang/frescodemo/MainActivity.java	(revision c1ff51a2a455aa259ee75934cce72ea3aae60743)
+++ app/src/main/java/com/lihang/frescodemo/MainActivity.java	(revision )
@@ -1,13 +1,21 @@
 package com.lihang.frescodemo;
 
+import android.graphics.drawable.Animatable;
 import android.net.Uri;
+import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 
+import com.facebook.common.logging.FLog;
 import com.facebook.drawee.backends.pipeline.Fresco;
 import com.facebook.drawee.backends.pipeline.PipelineDraweeController;
+import com.facebook.drawee.controller.BaseControllerListener;
+import com.facebook.drawee.controller.ControllerListener;
 import com.facebook.drawee.interfaces.DraweeController;
 import com.facebook.drawee.view.SimpleDraweeView;
+import com.facebook.imagepipeline.image.ImageInfo;
+import com.facebook.imagepipeline.image.QualityInfo;
 import com.facebook.imagepipeline.request.ImageRequest;
 import com.facebook.imagepipeline.request.ImageRequestBuilder;
 import com.lidroid.xutils.ViewUtils;
@@ -24,24 +32,76 @@
     private String img_url3 = "http://hiphotos.baidu.com/%CC%EC%C9%BD%B6%FE%CF%C0%B5%C4%D0%A1%CE%DD/pic/item/70c553e736d12f2e5b0614d64fc2d5628535682a.jpg";
     private String img_url4 = "http://img4.douban.com/view/photo/raw/public/p1773847919.jpg";
     private String img_url5 = "http://7mno4h.com2.z0.glb.qiniucdn.com/3859deb07d4647cf9183f8ea3f5aa165.jpg";//大图
-    private String img_url6 = img_url5+"/test";//小图
+    private String img_url6 = img_url5 + "/test";//小图
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         ViewUtils.inject(this);
-        requestImage();
 //        Uri uri = Uri.parse(img_url0);
 //        simpleDraweeView.setImageURI(uri);
+//        requestImage();
+        moreImages();
+
+
     }
 
-    private void requestImage(){
+    private void requestImage() {
-        ImageRequest request = ImageRequestBuilder.newBuilderWithSource( Uri.parse(img_url4))
+        ImageRequest request = ImageRequestBuilder.newBuilderWithSource(Uri.parse(img_url4))
                 .setProgressiveRenderingEnabled(true)
                 .build();
-            PipelineDraweeController controller = (PipelineDraweeController) Fresco.newDraweeControllerBuilder()
+        PipelineDraweeController controller = (PipelineDraweeController) Fresco.newDraweeControllerBuilder()
                 .setImageRequest(request)
+                .setOldController(simpleDraweeView.getController())
+                .build();
+        simpleDraweeView.setController(controller);
+    }
+
+    private void moreImages() {
+        ControllerListener controllerListener = new BaseControllerListener<ImageInfo>() {
+            //成功
+            @Override
+            public void onFinalImageSet(
+                    String id,
+                    @Nullable ImageInfo imageInfo,
+                    @Nullable Animatable anim) {
+                if (imageInfo == null) {
+                    return;
+                }
+                QualityInfo qualityInfo = imageInfo.getQualityInfo();
+                FLog.d("Final image received! " +
+                                "Size %d x %d",
+                        "Quality level %d, good enough: %s, full quality: %s",
+                        imageInfo.getWidth(),
+                        imageInfo.getHeight(),
+                        qualityInfo.getQuality(),
+                        qualityInfo.isOfGoodEnoughQuality(),
+                        qualityInfo.isOfFullQuality());
+                Log.e("==========>>","Final image received! " +
+                                "Size "+imageInfo.getWidth()+" x "+imageInfo.getHeight()+
+                        "Quality level "+qualityInfo.getQuality()+", good enough: "+qualityInfo.isOfGoodEnoughQuality()+", full quality: "+qualityInfo.isOfFullQuality());
+            }
+
+            //渐进
+            @Override
+            public void onIntermediateImageSet(String id, @Nullable ImageInfo imageInfo) {
+                FLog.d("", "Intermediate image received");
+                Log.e("=========>>","Intermediate image received");
+            }
+
+            //失败
+            @Override
+            public void onFailure(String id, Throwable throwable) {
+                FLog.e(getClass(), throwable, "Error loading %s", id);
+                Log.e("", "Error loading "+id);
+            }
+        };
+
+        DraweeController controller = Fresco.newDraweeControllerBuilder()
+                .setControllerListener(controllerListener)
+                .setLowResImageRequest(ImageRequest.fromUri(img_url6))
+                .setImageRequest(ImageRequest.fromUri(img_url5))
                 .setOldController(simpleDraweeView.getController())
                 .build();
         simpleDraweeView.setController(controller);
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/misc.xml	(revision c1ff51a2a455aa259ee75934cce72ea3aae60743)
+++ .idea/misc.xml	(revision )
@@ -37,7 +37,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
\ No newline at end of file
Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(revision c1ff51a2a455aa259ee75934cce72ea3aae60743)
+++ app/src/main/res/layout/activity_main.xml	(revision )
@@ -8,9 +8,10 @@
 
     <com.facebook.drawee.view.SimpleDraweeView
         android:id="@+id/iv_image_view"
-        android:layout_width="130dp"
-        android:layout_height="130dp"
+        android:layout_width="300dp"
+        android:layout_height="180dp"
         fresco:placeholderImage="@mipmap/ic_launcher"
+        android:layout_centerInParent="true"
         />
 
 </RelativeLayout>
